DuckDuckGo/TabDelegate.swift

```diff
@@ -22,6 +22,8 @@ import Core
 
 protocol TabDelegate: class {
 
+    func tabWillRequestNewTab(_ tab: TabViewController) -> TabViewController.LinkDestination
```

Comment:
@bwaresiak This is now handled by having `MainViewController` keep track of the tap link destination state, and `TabViewController` asks for the destination via this new protocol method. I want to avoid having `TabViewController` knowing about `MainViewController` directly, so this seems better even if this function is very simple. 🙂

Reply:
How about removing the `LinkDestination` enum and replacing that directly with `UIKeyModifierFlags` in MainVC, then move the logic of "unwrapping" flags in the place that is supposed to interpret these (navigation)?

Reply:
That's even better :+1: That approach will mean that `MainViewController` isn't trying to interpret what the flags are supposed to mean itself, and should make it more generalized. Thanks! I'll push a commit shortly.